src/appleseed/renderer/modeling/surfaceshader/diagnosticsurfaceshader.cpp

```diff
@@ -583,7 +583,7 @@ void DiagnosticSurfaceShader::evaluate(
       case Assemblies:
         set_shading_result(
             shading_result,
-            integer_to_color3<float>(shading_point.get_assembly_instance().find_assembly()->get_uid()));
+            integer_to_color3<float>(shading_point.get_assembly_instance().get_assembly().get_uid()));
```

Comment:
Not what I suggested, please read again my comments :)

Reply:
ahh ahh ah sorry for that! fixing it now!

Reply:
done! sorry still getting used to the codebase :)